spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +75,21 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	public void setArguments(Map<Integer, Map<String, String>> arguments) {
+		this.arguments = arguments;
+	}
+
+	public Map<Integer, Map<String, String>> getArguments() {
+		return arguments;
+	}
+
+	public List<ModuleLaunchRequest> asModuleLaunchRequests() {
```

Comment:
I disagree, IMO this class alone should know about the details.
Moreover, the launcher typically deals with one module at a time, so would not know about the module index.

If we follow your recommendation, the the launcher is dealing with low level maps, and we can get rid of ModuleLaunchRequest